modules/core/src/main/scala/lucuma/odb/api/model/ElevationRangeModel.scala

```diff
@@ -0,0 +1,219 @@
+import io.circe.generic.semiauto.deriveDecoder
+import lucuma.core.math._
+import monocle.{ Getter, Prism }
+import monocle.macros.GenPrism
+
+/**
+ * Elevation range constraint. Can either be an AirmassRange or HourAngleRange.
+ */
+sealed trait ElevationRangeModel extends Product with Serializable
+
+object ElevationRangeModel {
+  val airmassRange: Prism[ElevationRangeModel, AirmassRange] =
+    GenPrism[ElevationRangeModel, AirmassRange]
+
+  val hourAngleRange: Prism[ElevationRangeModel, HourAngleRange] =
+    GenPrism[ElevationRangeModel, HourAngleRange]
+
+  implicit val EqElevationRange: Eq[ElevationRangeModel] = Eq.fromUniversalEquals
+
+  final case class Create(
+    airmassRange:   Option[AirmassRange.Create],
+    hourAngleRange: Option[HourAngleRange.Create]
+  ) {
+    def create: ValidatedInput[ElevationRangeModel] =
+      ValidatedInput.requireOne("elevationRange",
+                                airmassRange.map(_.create),
+                                hourAngleRange.map(_.create)
+      )
+  }
+
+  /**
+   * Input parameter used to create an elevation range. Both fields
+   * are optional, but validation will check that exactly one is defined.
+   */
+  object Create {
+    val Empty: Create = Create(airmassRange = None, hourAngleRange = None)
+
+    def airmassRange(amr: AirmassRange.Create): Create =
+      Empty.copy(airmassRange = amr.some)
+
+    def hourAngleRange(har: HourAngleRange.Create): Create =
+      Empty.copy(hourAngleRange = har.some)
+
+    implicit val DecoderCreate: Decoder[Create] = deriveDecoder
+
+    implicit val EqCreate: Eq[Create] =
+      Eq.by(c => (c.airmassRange, c.hourAngleRange))
+
+    implicit val InputValidatorCreate: InputValidator[Create, ElevationRangeModel] =
+      InputValidator.by(_.create)
+  }
+}
+
+/**
+ * Elevation range defined as an airmass range.
+ * deciMin must be <= deciMax
+ */
+sealed abstract case class AirmassRange protected (
+  deciMin: AirmassRange.IntDeciValue,
+  deciMax: AirmassRange.IntDeciValue
+) extends ElevationRangeModel {
+  def min: Double = deciMin.value / 10.0
+  def max: Double = deciMax.value / 10.0
+}
+
+object AirmassRange extends AirmassRangeOptics {
+  type DeciValue    = Interval.Closed[10, 30]
+  type IntDeciValue = Int Refined DeciValue
+
+  /**
+   * Construct a new AirmassRange.
+   * deciMin must be <= deciMax.
+   * @group Optics
+   */
+  def apply(deciMin: IntDeciValue, deciMax: IntDeciValue): Option[AirmassRange] =
+    fromOrderedDeciVals.getOption((deciMin, deciMax))
+
+  implicit val EqAirmassRange: Eq[AirmassRange] = Eq.fromUniversalEquals
+
+  /**
+   * Input parameter used to create an airmass range. Parameter ranges
+   * are validated and deciMin must be <= deciMax
+   */
+  final case class Create(deciMin: Int, deciMax: Int) {
+    def create: ValidatedInput[AirmassRange] =
+      (ValidatedInput.closedInterval("deciMin", deciMin, 10, 30),
+       ValidatedInput.closedInterval("deciMax", deciMax, 10, 30)
```

Comment:
A goal of the API is to make it possible to enter values in units that would seem most natural to the user.  See the `WavelengthModel` or `RightAscensionModel` for example.  In this case, my understanding is that astronomers will think of airmass in decimal values where `1.0` is the minimum (zenith).  If I understand it, we'd be asking them to convert this to 10 X the airmass for data entry?

Reply:
I was trying to keep exact values for the storage values. But, I guess BigDecimal would actually provide that for this situation. Entry in the UI would have been in decimals, and I was going to provide alternative units for the API. But, changing to BigDecimal, or even Double, would probably be sufficient for this case. (I just have a knee jerk reaction to floating point numbers.)

Reply:
Agree about the reaction to floating point and about storing them as you have it.  I think it's fine though to allow API users to send decimal values.  Though it is more work, what I've been doing is supporting all natural units.  The smallest possible useful resolution of something is specified as an integer/long value and others as `BigDecimal`.

Reply:
When it is translated into a GraphQL input, it will have required `deciMin` and `deciMax` parameters by default.  That'd be fine for Explore or other UIs but doesn't seem like it would be intuitive for most human users.  I hate to suggest it but I think we might want an `AirmassModel` along the lines of the `WavelengthModel` (with integer `deciAirmass` and decimal `airmass` options?) or else to just switch to `min` and `max` `BigDecimal` airmasses.  Same for hour angle below.

Reply:
I'm fine with creating something like the `WavelengthModel` for these. However, the degree of precision of BigDecimal might be OK in this case. The UI will only be allowing 1 decimal place. This doesn't seem like the kind of value we'll be doing a lot of repeated conversions with. Will we just be sending it to the Constraint Solver?

Reply:
Right, unlike wavelengths or angular separation there aren't multiple unit options that really make sense.   Making the values `BigDecimal` seems fine to me.